docs/proposals/20210310-opt-in-autoscaling-from-zero.md

```diff
@@ -188,6 +195,40 @@ type DockerMachineTemplate struct {
 ```
 _Note: the `ResourceList` and `ResourceName` referenced are from k8s.io/api/core/v1`_
 
+`platform.NodeInfo` is a struct that contains the architecture and operating system information of the node, to implement 
+in the providers integration code.
+Its definition would look like this:
+
+```go
+package platform
+
+// Architecture represents the CPU architecture of the node. Its underlying type is a string.
+// Its underlying type is a string and its value can be any of amd64, arm64, s390x, ppc64le.
+// +kubebuilder:validation:Enum=amd64;arm64;s390x;ppc64le
+// +enum
+type Architecture string
+
+// Architecture constants defined for clarity.
+const (
+    ArchitectureAmd64 Architecture = "amd64"
+    ArchitectureArm64 Architecture = "arm64"
+    ArchitectureS390x Architecture = "s390x"
+    ArchitecturePpc64le Architecture = "ppc64le"
+)
+
+// NodeInfo contains information about the node's architecture and operating system.
+type NodeInfo struct {
```

Comment:
I would suggest to use NodeSystemInfo, so there is a more direct link with the corresponding struct in corev1 types.

Reply:
Hi @fabriziopandini, the previous version was using NodeSystemInfo. I changed it due to https://github.com/kubernetes-sigs/cluster-api/pull/11962#discussion_r2018838962. cc @sbueringer 

I'm ok with either way as far as we have consensus :).

Reply:
I would favor NodeInfo because I think the field and struct name should be the same. And I think the field name should be the same as in upstream

Reply:
I'm ok with both options, but the consistency with corev1.Node in this case IMO could help

Reply:
I agree with both your points, but leaning a bit more into @fabriziopandini's option to be consistent with corev1.Node.

Reply:
I was thinking about this a little bit more and considering those are type implemented by each provider, I think we should not be prescriptive about the type name nor host those types in CAPI at all.

Instead define in the contract only what the implementations will rely on, like e.g. we are doing for https://main.cluster-api.sigs.k8s.io/developer/providers/contracts/infra-machine#inframachine-conditions
(as you might notice, we are only defining what we expect will end up in the K8s8 resource / in etcd, without being prescriptive on go modeling).

Reply:
> I'll be happy to contribute the changes to the book as well once this merges, if you're ok with it.

I don't mind if now or later, but documentation I would prefer to not defer after the implementation (so either now or in the implementation PR).

Reply:
> Do you want me to remove line n. 203?

Yes please remove the package name, this could be misleading